src/gc/gc.d

```diff
@@ -3132,19 +3134,17 @@ struct SmallObjectPool
 
                 rt_finalizeFromGC(q, size, attr);
 
-                toClear |= GCBits.BITS_1 << clearIndex;
+                freeBits = true;
+                toFree.set(i);
 
                 debug(COLLECT_PRINTF) printf("\tcollecting %p\n", p);
                 //log_free(sentinel_add(p));
 
                 debug (MEMSTOMP) memset(p, 0xF3, size);
-                freebits.set(biti);
```

Comment:
Slightly OT, but I recently wondered why we have to set freebits in runFinalizers that is called from outside of the GC (or any other GCBits apart from final). Outside of a collection, this does nothing (so far), while the item should be sent to the bucket list if it is meant to be freed (which is what the large object pool version does).

For safer operation, I think runFinalizers should not free the allocation, just finalize them so that the virtual function table pointer is cleared.